### PR TITLE
Make some other nullable parameters explicitly nullable

### PR DESCRIPTION
**Pull request by jorgeatorres**
Here we continue the work started in #1205, where most function parameters that are implicitly nullable were explicitly marked as nullable. That is, function definitions such as 

```
public function __construct( ActionScheduler_Store $store = null, $batch_size = 20 )
```

were turned into

```
public function __construct( ?ActionScheduler_Store $store = null, $batch_size = 20 ) {
```

(Notice the type change in the `$store` argument).

We missed a couple instances of these nullable args in the prior PR and so that's what this one adds. The aim is to reduce noise when running A-S (or a plugin that bundles A-S) under PHP 8.4.

### CODE REVIEW

File: lib/cron-expression/CronExpression.php
Comment (coreymckrill) [inline lib/cron-expression/CronExpression.php:1]: Given that this is a 3rd party library, I wondered if we should be directly editing the files in it. It looks like [it is abandoned](https://packagist.org/packages/mtdowling/cron-expression), which might be why we've got it under version control rather than as a composer dependency. At some point, the package [was forked](https://github.com/dragonmantank/cron-expression), and the new maintainer is still active. The latest version does add the [necessary nullable type](https://github.com/dragonmantank/cron-expression/blob/f37e405332cd1cca9d287f5044ae2a72c2f53239/src/Cron/CronExpression.php#L150), but the readme also says there are breaking changes since the version that we're using.

So in short, probably fine to edit this file, but perhaps we should look into updating this to a composer dependency using the latest version, both for PHP compatibility reasons, and also potentially security reasons.

**Review by coreymckrill — APPROVED** [review]
👍 Did a quick smoke test and everything looks fine.

File: lib/cron-expression/CronExpression.php
Comment (coreymckrill) [reply lib/cron-expression/CronExpression.php:1]: Opened an issue for this: https://github.com/woocommerce/action-scheduler/issues/1221

Comment (jorgeatorres) [reply lib/cron-expression/CronExpression.php:1]: Thank you @coreymckrill! 💯 Same thought crossed my mind when I was modifying that file but forgot to mention it in the description.
